service/src/java/org/apache/hive/service/cli/TezProgressMonitorStatusMapper.java

```diff
@@ -0,0 +1,36 @@
+package org.apache.hive.service.cli;
+
+import org.apache.hive.service.rpc.thrift.TJobExecutionStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TezProgressMonitorStatusMapper implements ProgressMonitorStatusMapper {
+  private final Logger LOG = LoggerFactory.getLogger(TezProgressMonitorStatusMapper.class.getName());
+
+  /**
+   * These states are taken form DAGStatus.State, could not use that here directly as it was
+   * optional dependency and did not want to include it just for the enum.
+   */
+  enum TezStatus {
+    SUBMITTED, INITING, RUNNING, SUCCEEDED, KILLED, FAILED, ERROR
+
+  }
+
+  @Override
+  public TJobExecutionStatus forStatus(String status) {
+    try {
+      TezStatus tezStatus = TezStatus.valueOf(status);
+      switch (tezStatus) {
+      case SUBMITTED:
+      case INITING:
+      case RUNNING:
+        return TJobExecutionStatus.IN_PROGRESS;
+      default:
+        return TJobExecutionStatus.COMPLETE;
+      }
+    } catch (IllegalArgumentException ex) {
+      LOG.debug("Could not understand the status from tez execution engine : " + status);
```

Comment:
Was done as debug explicitly, as when we disable hive server two in place logging the mapper will still try to get the corresponding status which it will not find because status string will be "". This is only to be logged explicitly when we want to debug a case, the exception here does not indicate any sort of problem with execution  / internal state etc.